### DISCUSSION TITLE
[CLARIFICATION REQUIRED] Issues with pulling private docker images from dockerhub

### DISCUSSION THREAD

**Discussion by dodalovic** [k3s-io/k3s#5521] [General]
Hi, thanks for the tool, first of all. 
I'm trying to configure k3s to pull docker images from the docker hub. 

I saw some examples, where one can just create something like this:

![image](https://user-images.githubusercontent.com/6358061/165636837-46d883c8-65cb-4514-871c-c65705f91627.png)

and then one can mount such secret into pod, and the images should be pulled. 

On the other side, I see https://rancher.com/docs/k3s/latest/en/installation/private-registry/ where it says I need to create some file... I'm confused. 

Thanks

**Comment by brandond** [comment]
The first thing you're looking at is called Image Pull Secrets. If the image is on a registry that requires authentication, you can store the credentials in a secret, and reference the secret from the pod, to use them when pulling the image.
https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/

If you want to provide credentials at a container runtime level (without having to configure each and every pod), or if you want to configure a system-level registry mirror, then use the k3s private registry configuration.

**Reply by dodalovic** [reply]
Hi, @brandond, first of thanks that you have time for this question at all. 

Suppose I want to use this system level setup, and I want to pull my private images from docker hub:

Do I need something like:

```shell
cat /etc/rancher/k3s/registries.yaml
mirrors:
  docker.io:
    endpoint:
      - "https://myreg.com:5000"
configs:
  "myreg.com:5000":
    auth:
      username: my-username
      password: my-password
```

And then just refernce my images normally in the helm template, `dodalovic/whatever:latest` or ?
